salt/transport/zeromq.py

```diff
@@ -838,15 +844,14 @@ def _publish_daemon(self, log_queue=None):
                     raise exc
 
         except KeyboardInterrupt:
-            # Cleanly close the sockets if we're shutting down
-            if pub_sock.closed is False:
-                pub_sock.setsockopt(zmq.LINGER, 1)
-                pub_sock.close()
-            if pull_sock.closed is False:
-                pull_sock.setsockopt(zmq.LINGER, 1)
-                pull_sock.close()
-            if context.closed is False:
-                context.term()
+            log.trace('Publish daemon caught Keyboard interupt, tearing down')
+        # Cleanly close the sockets if we're shutting down
+        if pub_sock.closed is False:
+            pub_sock.close()
+        if pull_sock.closed is False:
+            pull_sock.close()
+        if context.closed is False:
+            context.term()
```

Comment:
Does the above need to be indented... part of the KeyboardInterrupt error?

Reply:
Instead of only cleaning up the sockets and context on a keyboard error. I moved this so it will always run when the daemon process is ending.

Reply:
Sweet